sim-rs/LINEAR_LEIOS.md

```diff
@@ -0,0 +1,30 @@
+# Linear Leios (rust simulation)
+
+To run it, set `leios-variant` to `linear`.
+
+The log file schema is currently identical to every other variant (though `pipeline` is always 0).
+
+## Description
+
+Whenever a node creates an RB, it also creates an EB. The RB header contains a reference to this new EB. If the RB producer has a certificate for the parent RB’s EB, it will include that certificate in the RB body.
+
+RB headers are diffused separately from bodies. When a node receives an RB header, it checks whether that RB should be the new head of its chain. If so, it will request the RB body and the referenced EB (from the first peer which announces them).
+
+When a node receives an RB body, it immediately removes all referenced/conflicting transactions from its mempool. If the RB has an EB certificate, it also removes that EB’s transactions from its mempool. 
+
+When a node receives an EB body, it immediately runs a VRF lottery, and if successful, transmits votes for that EB. If the EB has been certified after L_vote + L_diff slots have passed, the node removes all of its transactions from the mempool (under the assumption that the EB will make it on-chain).
```

Comment:
> `it immediately runs a VRF lottery`

Could you elaborate on this part? Maybe it's merely checking whether the voting leader schedule elects it for the slot of the EB?

Reply:
When a node sees an EB, it runs a (simulated) VRF lottery to decide how many votes it has the right to produce. I don't know how VRF lotteries work for real, but if they have to be tied to a slot then we could use the slot of the EB for that (and I guess a node just has the right to vote for all EBs produced in a slot if there are slot battles).

Totally possible that I'm abusing the term or concept of VRF lottery here. my cryptographic background is pretty week.